search/2015-02-28-Preview/swagger/searchindex.json

```diff
@@ -0,0 +1,280 @@
+        }
+      },
+      "description": "Status of an indexing operation for a single document.",
+      "x-ms-external": true
+    },
+    "DocumentIndexResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "readOnly":  true,
+          "items": {
+            "$ref": "#/definitions/IndexingResult"
+          },
+          "description": "Gets the list of status information for each document in the indexing request."
+        }
+      },
+      "description": "Response containing the status of operations for all documents in the indexing request.",
+      "x-ms-external": true
+    },
+    "IndexActionType": {
+      "type": "string",
+      "enum": [
+        "upload",
+        "merge",
+        "mergeOrUpload",
+        "delete"
+      ],
+      "x-ms-enum": { "name": "IndexActionType" },
+      "description": "Specifies the operation to perform on a document in an indexing batch."
+    },
+    "SearchMode": {
+      "type": "string",
+      "enum": [
+        "any",
+        "all"
+      ],
+      "x-ms-enum": { "name": "SearchMode" },
+      "description": "Specifies whether any or all of the search terms must be matched in order to count the document as a match."
+    },
+    "QueryType": {
+      "type": "string",
+      "enum": [
+        "simple",
+        "full"
+      ],
+      "x-ms-enum": { "name": "QueryType" },
+      "description": "Specifies the syntax of the search query. The default is 'simple'. Use 'full' if your query uses the Lucene query syntax."
+    },
+    "SearchParameters": {
+      "properties": {
+        "filter": {
+          "externalDocs": {
+            "url": "https://msdn.microsoft.com/library/azure/dn798921.aspx"
+          },
+          "type": "string",
+          "description": "Gets or sets the OData $filter expression to apply to the search query."
+        },
+        "highlightFields": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Gets or sets the list of field names to use for hit highlights. Only searchable fields can be used for hit highlighting."
+        },
+        "highlightPostTag": {
+          "type": "string",
+          "description": "Gets or sets a string tag that is appended to hit highlights. Must be set with HighlightPreTag. Default is &lt;/em&gt;."
+        },
+        "highlightPreTag": {
+          "type": "string",
+          "description": "Gets or sets a string tag that is prepended to hit highlights. Must be set with HighlightPostTag. Default is &lt;em&gt;."
+        },
+        "includeTotalResultCount": {
+          "type": "boolean",
+          "description": "Gets or sets a value that specifies whether to fetch the total count of results. Default is false. Setting this value to true may have a performance impact. Note that the count returned is an approximation."
```

Comment:
wherever the default is specified, it may be useful to specify the default. That value will be passed/set in the request. If this purely to let the user know, then it should be fine to not specify them.